### PR TITLE
Add more functions for retrieving incidents and incident notes

### PR DESCRIPTION
**Pull request by arunasank**
cc/ @batpad @ianshward 

TODO:
* [x] Pass PagerDuty URL in index.js, and check to see if all ✅ - @arunasank

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -40,54 +42,96 @@ pagerduty.prototype.stream = function(status, services, interval) {
...
-        var ids = _(services).reduce(function(memo, service) {
-            if (_(names).indexOf(service.name) !== -1) memo.push(service.id);
-            return memo;
-        }, []);
-        if (!ids.length) return cb(new Error('No matching services found.'));
-        cb(null, ids);
     });
 };
+
+pagerduty.prototype.getIncidentNotes = function (ID, cb) {
```
Comment (ianshward) [inline index.js]: Is there a reason this is `ID` and not `id` ?

**Comment by arunasank** [comment]
Changed the `index.js` file so that the new calls work. @ianshward good to 🚢?

File: bin/incidents.js [hunk 2]
```diff
@@ -19,15 +19,17 @@ var argv = optimist
...
     })
-    .on('error', function(error) {
+    .on('error', function (error) {
+        console.log('got error');
         console.log(error);
     })
-    .on('end', function() {
-
+    .on('end', function () {
+        console.log('got end');
```
Comment (ianshward) [inline bin/incidents.js]: @arunasank is this debug output?

File: index.js [hunk 3]
```diff
@@ -1,34 +1,36 @@
+'use strict';
 var Readable = require('stream').Readable;
-var _ = require('underscore');
 var request = require('request');
 
-var pagerduty = function(subdomain, token) {
+var pagerduty = function (subdomain, token, url) {
```
Comment (ianshward) [inline index.js]: @arunasank is `subdomain` no longer used?  You could get rid of it if it's no longer used.

File: index.js [hunk 4]
```diff
@@ -1,34 +1,36 @@
...
         calling = true;
-        that.getIncidents(status, services, function(err, incidents) {
+        that.getIncidents(options, function (err, data) {
+            var incidents = data.incidents;
             if (err) return stream.emit('error', err);
             for (var i = 0; i < incidents.length; i++) {
                 var incident = incidents[i];
                 if (!seen[incident.id]) {
-                    stream.push(incident);
+                    stream.push(JSON.stringify(incident, null, 2));
```
Comment (ianshward) [inline index.js]: Is there a reason for stringifying here?  I'm just curious whether it is for convenience, or another reason.

**Comment by ianshward** [comment]
@arunasank I left a couple comments, and seeing that the README still needs to be updated to reflect the new constructor arguments.  It still says `var pd = new Pagerduty('mySubdomain', 'myToken');`

Otherwise, looks good to me.  Still no tests :sweat_smile: and I did not help w/ that situation since I didn't write any initally.  It might be a good point to add some though, while the code is fresh in your memory, and they could copy some of what you put together in the mbxcli branch (the mock PD server).

**Comment by ianshward** [comment]
> forgot to push tests!

LOL :spades: amazing

File: index.js (hunk 4 above)
Comment (arunasank) [inline index.js]: Just convenience. Can remove it.

**Comment by arunasank** [comment]
@ianshward Made some changes here. I think this will mean some minor changes in mbxcli, but was thinking that I'd make those once things get merged here so that there is not a lot of back and forth changes on code between here and there. Can you take a 👀 here when you find some time?

**Comment by ianshward** [comment]
@arunasank LGTM, and the plan sounds good.

**Comment by arunasank** [comment]
@ianshward Thanks! 🚢

**Pull request merged**